velox/common/base/AsyncSource.h

```diff
@@ -40,7 +40,7 @@ template <typename Item>
 class AsyncSource {
  public:
   explicit AsyncSource(std::function<std::unique_ptr<Item>()> make)
-      : make_(make) {}
+      : make_(std::move(make)) {}
```

Comment:
The change itself is good, but you still won't be able to capture `unique_ptr` inside a `std::function` because it is required to be copyable.  This only avoid copying of the `std::function` itself.